File: app/src/main/java/line/javafoundation/producer_consumer/ProducerAndConsumer.java
package line.javafoundation.producer_consumer;


/**
 * 网上代码 用于理解synchronized
 */
public class ProducerAndConsumer {
    public static void main(String[] args) {
        Storage<Phone> storage = new Storage<Phone>();
        new Thread(new Producer(storage)).start();
        new Thread(new Consumer(storage)).start();
    }
}

File: javabase/src/main/java/com/yin/line/javabase/producer_consumer/ProducerAndConsumer.kt
package line.javafoundation.producer_consumer

/**
 * 网上代码 用于理解synchronized
 */
object ProducerAndConsumer {
    @JvmStatic
    fun main(args: Array<String>) {
        val storage = Storage<Phone>()
        Thread(Producer(storage)).start()
        Thread(Consumer(storage)).start()
    }
}